src/Symfony/Component/Security/Core/Authentication/Token/PreAuthenticatedToken.php

```diff
@@ -49,10 +49,19 @@ public function __construct($user, $credentials, string $providerKey, array $rol
      * Returns the provider key.
      *
      * @return string The provider key
+     *
+     * @deprecated since 5.2, use getFirewallName() instead
      */
     public function getProviderKey()
     {
-        return $this->providerKey;
+        trigger_deprecation('symfony/security-core', '5.2', 'Method "%s" is deprecated, use "getFirewallName()" instead.', __METHOD__);
+
+        return $this->firewallName;
+    }
+
+    public function getFirewallName(): string
+    {
+        return $this->firewallName;
```

Comment:
This could be `return $this->getProviderKey()` to better keep BC, right?

Reply:
Great idea, fixed :)